### ISSUE TITLE
Step-by-step bear guide

### ISSUE THREAD

**Issue by Adrianzatreanu** [coala-analyzer/documentation#58]
_From @Makman2 on May 20, 2015 18:35_

The *writing-bears* guide should be improved. Some notes here and there, but the major aspect I have in mind:
A step-by-step guide that implements an actual bear of our coalib (in a simplified manner), so the user sees how to operate on files and do something meaningful. In my mind is the indentation bear, that should be fairly simple for one language (So we don't take the general LanguageDefinition feature in account or whatever).

One more idea:
- A little paragraph about global bears (as demonstration: A bear that calculates the MD5 hash - or any other hash-value - from all files)

_Copied from original issue: coala-analyzer/coala#561_

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on May 20, 2015 18:38_

I think we should take that LanguageDefinition into account, however it will be realized, because this is one of the key thoughts on writing bears.

Also we should make an example about dependent bears.

**Comment by Adrianzatreanu** [comment]
_From @Makman2 on May 20, 2015 18:52_

Maybe we should extend the simple bear with the LanguageDefinition feature. First use hardcoded values, and then introduce the bear-writer into our cool language mechanics :)

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on May 20, 2015 18:54_

sure something like that.

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on May 20, 2015 18:54_

lets just first figure out how to make language mechanics cool ;)

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on July 2, 2015 12:53_

I have a good example: PyLintBear - truly simple, truly useful.

**Comment by Adrianzatreanu** [comment]
_From @Makman2 on July 3, 2015 0:21_

Oh yeah right, this would be a perfect example :)

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on July 16, 2015 14:51_

blocked by the new bear design

**Comment by Adrianzatreanu** [comment]
_From @Makman2 on July 16, 2015 16:24_

You think we should wait for the new design? We could already make the guide and port it later^^

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on July 16, 2015 16:30_

we could do that but we need to make sure that the new tutorial is well
ported then, not half heartedly adapted.

2015-07-16 18:24 GMT+02:00 Makman2 <notifications@github.com>:

**Comment by Adrianzatreanu** [comment]
_From @Makman2 on July 16, 2015 16:34_

That's an argument, keep it for 0.3 alpha ^^

**Comment by Adrianzatreanu** [comment]
_From @Makman2 on November 8, 2015 22:21_

Moving this to 0.4, we don't get the new bear-ecosystem done for 0.3.

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on November 28, 2015 18:11_

Not milestone relevant and we don't know when the new bear design is goig to happen.

**Comment by Adrianzatreanu** [comment]
_From @Makman2 on February 12, 2016 22:11_

We have somehow a guide, now I don't know how to improve it...
@sils1297 closing this issue?

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on February 12, 2016 22:14_

@Makman2 no we need to revamp the bear writing tutorial. That's really important.

**Comment by Adrianzatreanu** [comment]
_From @AbdealiJK on February 24, 2016 5:47_

Do we need this still ? We have a pretty decent Writing_Bears.rst and now Linter_Bear.rst and Self_Correcting_bears.rst have been added.

PyLintBear is now a LinterBear, so that's not a good example to use
Maybe we could use LineCountingBear ?

Better description of what needs to done is required. Possibly split this into smaller newcomer tasks would be a good idea ?

**Comment by Adrianzatreanu** [comment]
_From @sils1297 on February 24, 2016 9:34_

I do think we should improve our original writing bears tutorial.

**Comment by Adrianzatreanu** [comment]
Working on this on the new docs, using the new linter API
